src/dowel/csv_output.py

```diff
@@ -48,6 +49,36 @@ def record(self, data, prefix=''):
                            'Did you change key sets after your first '
                            'logger.log(TabularInput)?'.format(
                                set(self._fieldnames), set(to_csv.keys())))
+                # close existing log file
```

Comment:
The warning above this line can be deleted now, since this case is handled elegantly. Please also delete any private code that becomes unused by that change.